### ISSUE TITLE
Fatal error building libcompiler_rt/divsc3.c for RISC-V purecap

### ISSUE THREAD

**Issue by bsdjhb** [CTSRD-CHERI/llvm-project#367]
For some reason I don't get a dump of the file and test-runner script to reproduce.

```
===> lib/libcompiler_rt (obj,all,install)
/usr/cheri-riscv/bin/clang -target riscv64-unknown-freebsd13.0 -target riscv64-unknown-freebsd13.0 --sysroot=/usr/obj/usr/home/john/work/cheri/git/cheribsd/riscv.riscv64c/tmp -B/usr/cheri-riscv/bin -g -O2 -pipe -march=rv64imafdcxcheri -mabi=l64pc128d   -fpic -fvisibility=hidden -DVISIBILITY_HIDDEN -I/usr/home/john/work/cheri/git/cheribsd/contrib/libcxxrt -g -mno-relax -MD  -MF.depend.divsc3.o -MTdivsc3.o -std=gnu99 -Wno-format-zero-length -fstack-protector-strong -Wsystem-headers -Werror -Wall -Wno-format-y2k -Wno-uninitialized -Wno-pointer-sign -Wno-error=pass-failed -Wno-error=misleading-indentation -Wno-empty-body -Wno-string-plus-int -Wno-unused-const-variable -Wno-tautological-compare -Wno-unused-value -Wno-parentheses-equality -Wno-unused-function -Wno-enum-conversion -Wno-unused-local-typedef -Wno-address-of-packed-member -Wno-switch -Wno-switch-enum -Wno-knr-promoted-parameter  -Qunused-arguments    -c /usr/home/john/work/cheri/git/cheribsd/contrib/llvm-project/compiler-rt/lib/builtins/divsc3.c  -o divsc3.o
fatal error: error in backend: Cannot select: t61: f32,ch = load<(load 4 from
      constant-pool)> t0, t60, undef:i64,
      home/john/work/cheri/git/cheribsd/contrib/llvm-project/compiler-rt/lib/builtins/divsc3.c:40:13
  t60: i64 = add t73, t64, home/john/work/cheri/git/cheribsd/contrib/llvm-project/compiler-rt/lib/builtins/divsc3.c:40:13
    t73: i64 = PseudoLLA TargetConstantPool:i64<[2 x float] [float 0.000000e+00, float 1.000000e+00]> 0
      t72: i64 = TargetConstantPool<[2 x float] [float 0.000000e+00, float 1.000000e+00]> 0
    t64: i64 = shl t70, Constant:i64<2>, home/john/work/cheri/git/cheribsd/contrib/llvm-project/compiler-rt/lib/builtins/divsc3.c:40:13
      t70: i64 = setcc t7, t76, setoeq:ch, home/john/work/cheri/git/cheribsd/contrib/llvm-project/compiler-rt/lib/builtins/divsc3.c:38:17
        t7: f32,ch = CopyFromReg t0, Register:f32 %19, home/john/work/cheri/git/cheribsd/contrib/llvm-project/compiler-rt/lib/builtins/divsc3.c:38:17
          t6: f32 = Register %19
        t76: f32,ch = load<(load 4 from constant-pool)> t0, t78, undef:iFATPTR128
          t78: iFATPTR128 = PseudoCLGC TargetConstantPool:iFATPTR128<float 0x7FF0000000000000> 0
            t77: iFATPTR128 = TargetConstantPool<float 0x7FF0000000000000> 0
          t75: iFATPTR128 = undef
      t49: i64 = Constant<2>
  t45: i64 = undef
In function: __divsc3
*** Error code 70
```

**Comment by arichardson** [comment]
Clang recently switched to no longer forking a separate cc1 process and instead doing it all within the same process. I wonder if that somehow broke crash recovery on some systems.

**Comment by bsdjhb** [comment]
I just got a different crash building the kernel, so I think it has to do with how a fatal error is handled perhaps?

**Comment by arichardson** [comment]
Ah I think the in-process cc1 might be causing report_fatal_error to not create a reproducer.
I saw a review upstream to change the exit() call to abort() which should make it work again.

**Comment by bsdjhb** [comment]
```
          ******** /usr/home/john/work/cheri/git/cheribsd/divsc3.c ********

float a;
void b() { a = __builtin_copysignf(a ? 1 : 0, 0); }
```

**Comment by jrtc27** [comment]
Fixed in 2ed12ba8b4 on dev (which will be the commit that persists once the RISC-V backend work is merged) and 0c2b640a93 (on cheri-riscv-rebased3 as a stop-gap until that time).
